model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -1487,48 +1448,33 @@ def run_corrector_step(
             offset_provider={},
         )
 
-        start_cell_lb_plus2 = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.CellDim) + 2
+        cell_domain = h_grid.domain(dims.CellDim)
+        edge_domain = h_grid.domain(dims.EdgeDim)
+        start_cell_lb_level_3 = self.grid.start_index(
+            cell_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_3)
         )
 
-        start_edge_lb_plus6 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim) + 6
+        start_edge_lb_level_7 = self.grid.start_index(
+            edge_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_7)
         )
 
-        start_edge_nudging_plus1 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.nudging(dims.EdgeDim) + 1
-        )
-        end_edge_local = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim)
-        )
+        start_edge_nudging_level_2 = self.grid.start_index(edge_domain(h_grid.Zone.NUDGING_LEVEL_2))
+        end_edge_local = self.grid.end_index(edge_domain(h_grid.Zone.LOCAL))
 
-        start_edge_lb_plus4 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim) + 4
-        )
-        end_edge_local_minus2 = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim) - 2
+        start_edge_lb_level_5 = self.grid.start_index(
+            edge_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_5)
         )
+        end_edge_second_halo_line = self.grid.end_index(edge_domain(h_grid.Zone.HALO_LEVEL_2))
 
-        start_edge_lb = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim)
-        )
-        end_edge_end = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.end(dims.EdgeDim)
-        )
+        start_edge_lb = self.grid.start_index(edge_domain(h_grid.Zone.LATERAL_BOUNDARY))
+        end_edge_end = self.grid.end_index(edge_domain(h_grid.Zone.END))
 
-        start_cell_lb = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.CellDim)
-        )
-        end_cell_nudging = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.nudging(dims.CellDim)
-        )
+        start_cell_lb = self.grid.start_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY))
```

Comment:
More consistency renaming
```suggestion
        start_cell_lateral_boundary = self.grid.start_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY))
```